src/data/sentencepiece_vocab.cpp

```diff
@@ -28,9 +32,12 @@ class SentencePieceVocab : public VocabBase {
   Ptr<Options> options_;
   size_t batchIndex_{0};
 
+  std::mt19937 generator_;
```

Comment:
In other classes we use https://github.com/marian-nmt/marian-dev/blob/master/src/data/rng_engine.h It has been introduced for serialization of the seed. Maybe we can use it here as well?